src/Controls/tests/BindingSourceGen.UnitTests/BindingRepresentationGenTests.cs

```diff
@@ -290,4 +320,188 @@ class Foo
+        }
+        """;
+
+        var actualBinding = SourceGenHelpers.GetBinding(source);
+        var expectedBinding = new CodeWriterBinding(
+                new SourceCodeLocation("", 3, 7),
+                new TypeDescription("global::Foo"),
+                new TypeDescription("string"),
+                [
+                    new Cast(
+                        new MemberAccess("Value"),
+                        new TypeDescription("string")
+                        ),
+                ],
+                GenerateSetter: true
+            );
+
+        //TODO: Change arrays to custom collections implementing IEquatable
+        Assert.Equal(expectedBinding.Path, actualBinding.Path);
+        Assert.Equivalent(expectedBinding, actualBinding, strict: true);
+    }
+
+    [Fact]
+    public void GenerateBindingWhenGetterContainsMemberAccessOfCastReferenceType()
+    {
+        var source = """
+        using Microsoft.Maui.Controls;
+        var label = new Label();
+        label.SetBinding(Label.RotationProperty, static (Foo f) => (f.C as C).X);
+
+        public class Foo
+        {
+            public object C { get; set; }
+        }
+
+        class C
+        {
+            public int X { get; set; }
+        }
+        """;
+
+        var actualBinding = SourceGenHelpers.GetBinding(source);
+        var expectedBinding = new CodeWriterBinding(
+                new SourceCodeLocation("", 3, 7),
+                new TypeDescription("global::Foo"),
+                new TypeDescription("int", IsValueType: true),
+                [
+                    new Cast(
+                        new MemberAccess("C"),
+                        new TypeDescription("global::C")
+                        ),
+                    new MemberAccess("X"),
+                ],
+                GenerateSetter: true
+            );
+
+        //TODO: Change arrays to custom collections implementing IEquatable
+        Assert.Equal(expectedBinding.Path, actualBinding.Path);
+        Assert.Equivalent(expectedBinding, actualBinding, strict: true);
+    }
+
+    [Fact]
+    public void GenerateBindingWhenGetterContainsMemberAccessOfCastNullableReferenceType()
+    {
+        var source = """
+        using Microsoft.Maui.Controls;
+        var label = new Label();
+        label.SetBinding(Label.RotationProperty, static (Foo f) => (f.C as C)?.X);
+
+        public class Foo
+        {
+            public object? C { get; set; }
+        }
+
+        class C
+        {
+            public int X { get; set; }
+        }
+        """;
+
+        var actualBinding = SourceGenHelpers.GetBinding(source);
+        var expectedBinding = new CodeWriterBinding(
+                new SourceCodeLocation("", 3, 7),
+                new TypeDescription("global::Foo"),
+                new TypeDescription("int", IsNullable: true, IsValueType: true),
+                [
+                    new Cast(
+                        new ConditionalAccess(new MemberAccess("C")),
```

Comment:
Shouldn't the `Cast` and `ConditionalAccess` be nested other way around?
```c#
new ConditionalAccess(new Cast(new MemberAccess("C"), new TypeDescription("global::C")))
```
... as in - there's no conditional access to `C`, but it might not be of type `global::C` and the cast value might be `null`.

Consider this scenario:
```c#
static (Foo f) => (f?.C as C)?.X
```
I think that should end up as
```c#
new ConditionalAcces(new Cast(new ConditionalAccess(new MemberAccess(...)), ...))
```

Reply:
Hmm, not sure what you meant by `ConditionalAccess` on the second thought.

Is this the definition:  I try to access sth, but the thing Im trying to access might be null. If it's null then I return null. 
So basically it informs us that the object we try to access some property of can be null. And it does not matter whether the property is null or not?

If this is right then e. g.:

```csharp
Assert(foo!=null)
var result = foo.Bar; // This is just MemberAccess as f is not null, even if the result is actually null
      
class Foo
{
    Bar? Bar {get; set;}
}
```

And the lambda you gave:
```csharp
static (Foo f) => (f?.C as C)?.X
```
In my opinion then becomes:

```csharp
new ConditionalAccess(new MemberAccess(new Cast(new ConditionalAccess(new MemberAccess(...)), ...)))
```

So the `Cast` should not be wrapped in `ConditionalAccess` but rather the following `MemberAccess`

Reply:
Earlier I did consider the old `isNullable` parameter as information whether the underlying part can take null values. 

So in:
```csharp
static (Foo f) => f.Bar)

class Foo
{
    Bar? Bar {get; set;}
}
```
the part holding string `Bar` would be marked as `IsNullable` True

Reply:
> So basically it informs us that the object we try to access some property of can be null. 

Yes, exactly. The thing that could be null is in front of the `?` in the c# code. It is possible that the way I changed it is now different from the syntax tree and if that's the case, I probably did it the wrong way and we should align this again.

Reply:
I guess we need to align ourselves if "conditional access" in `a?.B` should mean "conditional access to `a` + member access to B` or `a + conditional access to B`.

Reply:
I do think that `ConditionalAccess` should mean option 2 (a + conditional access to B). 

I guess it becomes a bit weird in a situation we have nullable disabled. We could argue that accessing any property of reference type or nullable value type is `ConditionallyAccess` then. But then while parsing a syntax tree to determine whether a given MemberAccess is ConditionalAccess we would have to investigate the type of the expression on its left.

Reply:
I slept on it and I now agree with you 😄 https://github.com/simonrozsival/maui/pull/13

Reply:
Also there shouldn't definitely be two nested `ConditionalAccess` parts:
```c#
static (Foo f) => (f?.C as C)?.X
```
should be
```c#
[
    new Cast(new ConditionalAccess(new MemberAccess("C")), ...),
    new ConditionalAccess(new MemberAccess("X")),
]
```

Reply:
Yup no nested ConditionalAccess, this looks right

Reply:
So on the `nullable disabled` topic: do we mark every member access on reference types and nullable value types as `ConditionalAccess` ?

Reply:
> So on the nullable disabled topic

Yes, I think so. We'll need to carefully generate the C# code in a `#nullable enabled` context.